### DISCUSSION TITLE
Searching or changing difficulty temporarily freezing game

### DISCUSSION THREAD

**Discussion by xentfy** [ppy/osu#26108] [Q&A]
I have a lot of maps and when i type in the search bar or add a mod, the game can freeze for around 5-15 seconds. Wondering if theres a way to change the search algorithm to improve performance when searching for a map. Issue is only on lazer and not stable.

**Comment by peppy** [comment]
@xentfy if you set your song select sort mode to "artist" does it get considerably faster? i recently optimised this but had to turn off the optimisation for difficulty sort temporarily, because it was causing some issues.

also, how many beatmaps do you have imported?

**Comment by xentfy** [comment]
Hi Peppy, I have 106,680 maps imported. Switching to artist does search much faster and does not stutter, thank you for the suggestion.
